client/packages/coldchain/src/Equipment/DetailView/DetailView.tsx

```diff
@@ -73,11 +74,17 @@ export const EquipmentDetailView = () => {
 
   useEffect(() => {
     if (!data) return;
+    let properties = JSON.parse(data.properties);
+    if (!properties || !isObject(properties)) {
+      // Invalid properties, reset to empty object
+      properties = {};
+    }
     setDraft({
       ...data,
       locationIds: draft?.locationIds
         ? draft.locationIds
         : data.locations.nodes.map(location => location.id),
+      parsedProperties: properties,
```

Comment:
the reason for renaming the prop is to get the different type is it?

Reply:
Yes, do you have a better idea? Slightly hacky I know...

Reply:
slightly hacky maybe, but I think the best approach. The use of the `draft` object is a bit wild west anyway